docs/source/conf.py

```diff
@@ -123,13 +123,46 @@
     AttributeDocumenter,
     Documenter,
     MethodDocumenter,
+    PropertyDocumenter,
 )
 from sphinx.ext.autosummary import Autosummary  # isort:skip
 
 
+class ModinAccessorDocumenter(PropertyDocumenter):
+    """
+    Generates documentation for properties of Modin objects like Series.str and Series.dt that
+    are themselves accessor classes.
+    This class is necessary because we need to monkeypatch the Series.str/dt property objects
+    with the actual classes in order for autosummary-generate to produce stubs for them.
+
+    This class is not responsible for properties of those accessors like Series.str.capitalize.
+
+    See sphinx source for PropertyDocumenter:
+    https://github.com/sphinx-doc/sphinx/blob/907d27dc6506c542c11a7dd16b560eb4be7da5fc/sphinx/ext/autodoc/__init__.py#L2691
+    """
+
+    objtype = "modinaccessor"
+    directivetype = "attribute"
+
+    # lower priority than the other custom attribute documenter (defined below)
+    priority = 0.55
```

Comment:
what is this priority used for? why do we need this?

Reply:
I'm not sure why, but for some reason sphinx decides to check every single `Documenter` subclass with the specified `directivetype` (in this case "attribute") to determine which to use when documenting an attribute. The priority value (1 for the built-in default one) determines which class is used.

Reply:
i think I am still kind of fuzzy here, so if you do not specify a priority here, what will happen? can you give me an example here?

Reply:
If no priority is specified, then this custom class will be used when attempting to document any class's attributes. When I try removing the priority field and rebuilding, it attempts to use this class to load documentation for `DataFrame.columns`, and errors out because this class is hard-coded to work specifically for Series.

Since we want to use the default sphinx class in as many places as possible, we set this priority to be below that of the default, and below that used for sub-properties like `Series.str.capitalize`. This way, this custom class is only used when a method's template explicitly specifies the `modinaccessor::` directive.